cirq-core/cirq/sim/act_on_args_container_test.py

```diff
@@ -226,6 +227,12 @@ def test_copy_succeeds():
     assert copied.qubits == (q0, q1)
 
 
+def test_copy_deprecation_warning():
+    args = create_container(qs2, False)
+    with cirq.testing.assert_deprecated('reuse_buffer', deadline='0.25'):
+        args.copy(True)
```

Comment:
Showing my python ignorance here, but how/why does this call work? `EmptyActOnArgs.copy` doesn't accept an argument.

Reply:
It is because `args` is a `ActOnArgsContainer` and in `ActOnArgsContainer.copy`:
```
            if 'reuse_buffer' in inspect.signature(act_on_args.copy).parameters:
                copies[act_on_args] = act_on_args.copy(reuse_buffer)
            else:
                warnings.warn(
                    (
                        'A new parameter reuse_buffer has been added to ActOnArgs.copy(). The '
                        'classes that inherit from ActOnArgs should support it before Cirq 0.15.'
                    ),
                    DeprecationWarning,
                )
                copies[act_on_args] = act_on_args.copy()
```
So, it does not directly call `EmptyActOnArgs.copy`.